solr/core/src/java/org/apache/solr/store/blob/client/LocalStorageClient.java

```diff
@@ -97,7 +117,7 @@ public void pushCoreMetadata(String sharedStoreName, String blobCoreMetadataName
 
       // Writing to the file assumed atomic, the file cannot be observed midway. Might not hold here but should be the case
       // with a real S3 implementation.
-      try (PrintWriter out = new PrintWriter(blobMetadataFile)){
+      try (PrintWriter out = new PrintWriter(blobMetadataFile, StandardCharsets.UTF_8.name())){
```

Comment:
@andyvuong I've made an assumption here that all of our blob files are UTF-8 encoded, is that a valid assumption? We need to specify a charset to decode in several places

Reply:
This only affects core.metadata and not all blob files btw - local storage client is only used for testing purposes so this is fine here too.